### PR TITLE
test(cli): :test_tube: implement ci parser class to parse ci/cd runner information

### PR DESCRIPTION
**Pull request by EricRootLee**
*Replace this paragraph with a description of what this PR is changing or adding and why.*

### List of issues that are fixed by the PR
*You must list at least one issue.*

 - I removed test coverage in the widgetbook_cli workflow
   - I added Melos to the workflow to make the test pass
 - I Added  CiParser implementation for 
    - GitHub
    - Gitlab
    - Azure Pipelines
    - Bitbucket
- I added test for  Ci Parser abstract class
- I added test for CiParser Implementation for different vendors
- I added test Widgetbook Command Runner
- I added test for  Update Command
- I added test for  Publish Command
 
  

### Screenshots
*If applicable, add screenshots to help explain the changes.*

### Checklist

- [ ] I signed the [CLA].
- [ ] I listed at least one issue that this PR fixes in the description above.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [ ] I added new tests to check the change I am making].
- [ ] All existing and new tests are passing.

If you need help, consider asking for advice on [Discord].

[CLA]: https://docs.google.com/forms/d/e/1FAIpQLScuRfjUzENsLsmQgqZlGLxMKbFi7zuXoPARyXytoyQrq7ntUw/viewform
[Discord]: https://discord.com/invite/zT4AMStAJA

### CODE REVIEW

**Comment by EricRootLee** [comment]
Issue:

1. Code analysis Failing  on the CI , but passing locally,
2. Code coverage not writing to lcov.info file

File: .github/workflows/widgetbook-cli.yaml [hunk 1]
```diff
@@ -1,8 +1,8 @@
- on: 
+on: 
   pull_request:
     paths:
       - "packages/widgetbook_cli/**"
-      - ".github/workflows/widgetbook_cli.yaml"
+      - ".github/workflows/widgetbook-cli.yaml"
   push:
     branches:
       - main
```
[the remark this answers is not in the document]
Comment (EricRootLee) [reply .github/workflows/widgetbook-cli.yaml:8]: `widgetbook_cli.yaml` is the old one ,  i renamed it to  `widgetbook-cli.yam` to be consisted with the other `.yaml` files

File: packages/widgetbook_cli/bin/commands/publish.dart [hunk 2]
```diff
@@ -110,7 +119,11 @@ class PublishCommand extends WidgetbookCommand {
   @override
   final String name = 'publish';
 
-  late CiParserRunner? ciParserRunner;
+  CiParserRunner? ciParserRunner;
```
[the remark this answers is not in the document]
Comment (EricRootLee) [reply packages/widgetbook_cli/bin/commands/publish.dart:122]: Missed that.

File: packages/widgetbook_cli/bin/commands/publish.dart [hunk 3]
```diff
@@ -191,36 +210,83 @@ class PublishCommand extends WidgetbookCommand {
...
+  @visibleForTesting
+  void deleteZip(File zip) {
+    zip.delete();
   }
 
-  Future<void> _publishBuilds({
+  File? getZipFile(Directory directory) =>
+      _widgetbookZipEncoder.encode(directory);
+
+  Future<Map<String, dynamic>?> uploadDeploymentInfo({
```
[the remark this answers is not in the document]
Comment (EricRootLee) [reply packages/widgetbook_cli/bin/commands/publish.dart:242]: All of the  methods are visible for testing, but I can also annotate them with ` @visibleForTesting` to make them look nite

File: packages/widgetbook_cli/test/utils/set_up_platform_environment.dart [hunk 4]
```diff
@@ -0,0 +1,8 @@
+import 'dart:io';
+
+Future<void> setUpPlatformEnvironment({
+  required String key,
+  required String value,
+}) async {
+  await Process.run('export', ["$key='$value'"]);
+}
```
[the remark this answers is not in the document]
Comment (EricRootLee) [reply packages/widgetbook_cli/test/utils/set_up_platform_environment.dart:8]: Not required

**Review by jenshor — APPROVED** [review]

**Pull request merged**